R/mod_utilization.R

```diff
@@ -0,0 +1,151 @@
+        minDate = ymd(floor_date(update_date - 335, unit = "month")),
+        dateFormat = "MM yyyy",
+        view = "month",
+        value = ymd(floor_date(update_date, unit = "month") - 1),
+        minView = "months",
+        addon = "none",
+        autoClose = TRUE,
+        width = '25%'
+      ),
+      width = 12
+    )), 
+    plotlyOutput("bedPlot"),
+    br(),
+    ui_row(
+      title = "Summary",
+      fluidRow(
+        bs4Dash::column(4, bs4Dash::infoBoxOutput(ns(
+          "infobox_bed"
+        ), width = '100%')),
+        bs4Dash::column(4, bs4Dash::infoBoxOutput(ns(
+          "infobox_unit"
+        ), width = '100%')),
+        bs4Dash::column(4, bs4Dash::infoBoxOutput(ns(
+          "infobox_utilization"
+        ), width = '100%'))
+      ),
+      width = 12
+    )
+  )
+}
+
+mod_utilization_server <- function(input, output, session){
+  ns <- session$ns
+  
+  output$header <- shiny::renderUI(server_header("Bed and Unit Utilization", 
+                                                 format(input$date_range, "%B %Y"), 
+                                                 shiny::tags$p(shiny::icon("exclamation-triangle", 
+                                                                           style = "display:inline-block; 
+                                                                           color: #ffc107;"), 
+                                                               "During this time, congregate facilities should 
+                                                               be aiming to deconcentrate. If this causes 
+                                                               fluctuations in Utilization, that is okay. 
+                                                               Please continue to keep your clients safe.", 
+                                                               status = "warning")))
+  
+  output$bedPlot <-
+    renderPlotly({
+      ReportEnd <- ymd(input$utilizationDate) 
+      ReportStart <- floor_date(ReportEnd, unit = "month") -
+        years(1) +
+        months(1)
+      ReportingPeriod <- interval(ReportStart, ReportEnd)
+      
+      Provider <- input$providerListUtilization
+      
+      bedPlot <- utilization_bed |> 
+        gather("Month",
+               "Utilization",
+               -ProjectID,
+               -ProjectName,
+               -ProjectType) |>
+        filter(ProjectName == Provider,
+               
+               mdy(Month) %within% ReportingPeriod) |>
+        mutate(
+          Month = floor_date(mdy(Month), unit = "month"),
+          Bed = Utilization,
+          Utilization = NULL
+        )
+      
+      unitPlot <- utilization_unit |> 
+        gather("Month",
+               "Utilization",
+               -ProjectID,
+               -ProjectName,
+               -ProjectType) %>%
+        filter(ProjectName == Provider,
+               mdy(Month) %within% ReportingPeriod) |>
```

Comment:
Month should already be a Date, if it's not, this should be done on the RmData end